weasyl/test/web/test_restricted_tags.py

```diff
@@ -0,0 +1,201 @@
+    # This is our journalid
+    journal = re.match(regex_pattern, resp.request.url).group(1)
+
+    """ Other users tests"""
+    # Other users cannot add tags which have been user-restricted
+    resp = app.post('/submit/tags', {'submitid': submission, 'tags': tag_user}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_user not in resp.html.find(id='tags-textarea')
+    resp = app.post('/submit/tags', {'charid': character, 'tags': tag_user}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_user not in resp.html.find(id='tags-textarea')
+    resp = app.post('/submit/tags', {'journalid': journal, 'tags': tag_user}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_user not in resp.html.find(id='tags-textarea')
+
+    # Other users cannot add tags which have been globally restricted
+    resp = app.post('/submit/tags', {'submitid': submission, 'tags': tag_global}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_global not in resp.html.find(id='tags-textarea')
+    resp = app.post('/submit/tags', {'charid': character, 'tags': tag_global}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_global not in resp.html.find(id='tags-textarea')
+    resp = app.post('/submit/tags', {'journalid': journal, 'tags': tag_global}, headers={'Cookie': tag_adder_cookie}).follow()
+    assert tag_global not in resp.html.find(id='tags-textarea')
+
+    """ Content owner tests """
+    # Content owners can add tags which are user-restricted
+    resp = app.post('/submit/tags', {'submitid': submission, 'tags': tag_user}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_user
+    resp = app.post('/submit/tags', {'charid': character, 'tags': tag_user}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_user
+    resp = app.post('/submit/tags', {'journalid': journal, 'tags': tag_user}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_user
+
+    # Content owners can add tags which are globally restricted
+    resp = app.post('/submit/tags', {'submitid': submission, 'tags': tag_global}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_global
+    resp = app.post('/submit/tags', {'charid': character, 'tags': tag_global}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_global
+    resp = app.post('/submit/tags', {'journalid': journal, 'tags': tag_global}, headers={'Cookie': owner_cookie}).follow()
+    assert resp.html.find(id='tags-textarea').string == tag_global
+
+    """ Moderators (or above) tests) """
+    # Moderators (or above) can add tags which are user-restricted
+    resp = app.post('/submit/tags', {'submitid': submission, 'tags': tag_user}, headers={'Cookie': director_cookie}).follow()
```

Comment:
Note to self: Manually clear these, since they do exist as the user added them before (now that I think about this, the tag needs to be in the ``added`` bucket in ``associate()`` for this to even have a snowball's chance in hell at not working (so the test is currently not testing what it should be testing)). I'll fix this glitch later. (Or, just make three different blacklisted tags, since that would work too.)